app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -244,255 +244,328 @@ export default class TransactionStateManager extends EventEmitter {
 
     // commit txMeta to state
     const txId = txMeta.id;
-    const txList = this.getFullTxList();
-    const index = txList.findIndex((txData) => txData.id === txId);
-    txList[index] = txMeta;
-    this._saveTxList(txList);
+    this.store.updateState({
+      transactions: {
+        ...this.store.getState().transactions,
+        [txId]: txMeta,
+      },
+    });
   }
 
   /**
-   * merges txParams obj onto txMeta.txParams use extend to ensure
-   * that all fields are filled
-   * @param {number} txId - the id of the txMeta
-   * @param {Object} txParams - the updated txParams
+   * SearchCriteria can search in any key in TxParams or the base
+   * TransactionMeta. This type represents any key on either of those two
+   * types.
+   * @typedef {TxParams[keyof TxParams] | TransactionMeta[keyof TransactionMeta]} SearchableKeys
    */
-  updateTxParams(txId, txParams) {
-    const txMeta = this.getTx(txId);
-    txMeta.txParams = { ...txMeta.txParams, ...txParams };
-    this.updateTx(txMeta, `txStateManager#updateTxParams`);
-  }
 
   /**
-   * normalize and validate txParams members
-   * @param {Object} txParams - txParams
+   * Predicates can either be strict values, which is shorthand for using
+   * strict equality, or a method that receives he value of the specified key
+   * and returns a boolean.
+   * @typedef {(v: unknown) => boolean | unknown} FilterPredicate
    */
-  normalizeAndValidateTxParams(txParams) {
-    if (typeof txParams.data === 'undefined') {
-      delete txParams.data;
-    }
-    // eslint-disable-next-line no-param-reassign
-    txParams = normalizeTxParams(txParams, false);
-    this.validateTxParams(txParams);
-    return txParams;
-  }
 
   /**
-   * validates txParams members by type
-   * @param {Object} txParams - txParams to validate
+   * Retrieve a list of transactions from state. By default this will return
+   * the full list of Transactions for the currently selected chain/network.
+   * Additional options can be provided to change what is included in the final
+   * list.
+   *
+   * Note: The strange Object.<string,any> below is to make vscode's typescript
```

Comment:
I gave up on this, and will remove this note. Setting opts type to {Object.<string,any>} forced intellisense to show all of the param details on hover, but then resulted in types being ignored (wtf) so removed it.